R/combined.R

```diff
@@ -442,3 +442,50 @@ spim_add_forecast_region <- function(samples, forecast_days) {
   samples
 
 }
+
+## This is adapted from mcstate::pmcmc_predict
+spim_pmcmc_predict <- function(object, steps, prepend_trajectories = FALSE,
+                               n_threads = NULL, seed = NULL) {
+
+  if (is.null(object$predict)) {
+    stop("mcmc was run with return_state = FALSE, can't predict")
+  }
+  if (length(steps) < 2) {
+    stop("At least two steps required for predict")
+  }
+  if (steps[[1]] != object$predict$step) {
+    stop(sprintf("Expected steps[1] to be %d", object$predict$step))
+  }
+  if (prepend_trajectories && is.null(object$trajectories)) {
+    stop(paste("mcmc was run with return_trajectories = FALSE,",
+               "can't prepend trajectories"))
+  }
+
+  state <- object$state
+  info <- object$info$info[[length(object$info$info)]]
+  index <- sircovid::lancelot_index(info)$state
+  model <- object$predict$filter$model
+  n_threads <- n_threads %||% object$predict$filter$n_threads
+
+  pars <- apply(object$pars, 1,
+                function(x) {
+                  p <- object$predict$transform(x)
+                  p[[length(p)]]$pars})
```

Comment:
Untested, but this might be clearer:

```suggestion
  pars <- apply(object$pars, 1, function(x)
                last(object$predict$transform(x))$pars)
```

However, I am always concerned about `apply`'s return value so better might be

```
  pars <- lapply(seq_len(nrow(object$pars)), function(i)
                last(object$predict$transform(pars[i, ]))$pars)
```

but not totally sure here

Reply:
I've switched to that, the output is a list so you're right that `lapply` makes the most sense)